components/workspacekit/cmd/rings.go

```diff
@@ -495,6 +495,18 @@ var ring1Cmd = &cobra.Command{
 		//nolint:errcheck
 		defer slirpCmd.Process.Kill()
 
+		client, err = connectToInWorkspaceDaemonService(ctx)
+		if err != nil {
+			log.WithError(err).Error("cannot connect to daemon")
+			return
+		}
+		_, err = client.SetupPairVeths(ctx, &daemonapi.SetupPairVethsRequest{Pid: int64(cmd.Process.Pid)})
```

Comment:
Please move `client.Close()` to line 503 and use something like `defer client.Close()`

Reply:
Is there any particular reason why you don't use defer in other places?
https://github.com/gitpod-io/gitpod/blob/e30502ed12142a4125e63fda868f09001d8a3b53/components/workspacekit/cmd/rings.go#L74-L85